### PR TITLE
Add humble CI

### PR DESCRIPTION
**Pull request by Cakem1x**
This PR adds a Github Action for CI, based upon actions from [ros-tooling](https://github.com/ros-tooling/action-ros-ci).

It will check that
* we have all dependencies specified
  * deps are installed via rosdep apt
  * for deps not available this way, we can add them to the `source_dependencies.yaml` file. Caveat: This will increase build times, since **all** packages in this repo will get build and tested.
* build of workspace succeeds
 * workspace consisting of `mesh_tools` and repos from `source_dependencies.yaml`
* test of workspace succeeds

Check out [this action log](https://github.com/naturerobots/mesh_tools/actions/runs/9097031721/job/25003919454), for example, if you want to see more details.

### CODE REVIEW

**Comment by aock** [comment]
> Current job runtime is 10min. I think we can improve this, e.g. via caching or by tweaking the parameters, but I suggest doing that at a later in a separate PR if the need arises.

Wow, that's long. But ok for now.

File: .github/workflows/noetic.yml
Comment (aock) [inline .github/workflows/noetic.yml:1]: Shouldn't we keep this for the noetic branch?

Comment (aock) [reply .github/workflows/noetic.yml:1]: ah lets just try it out

**Review by aock — APPROVED** [review]

File: .github/workflows/noetic.yml
Comment (Cakem1x) [reply .github/workflows/noetic.yml:1]: The `noetic` branch continues to have this workflow file.
I think GitHub Actions use the repo state with which the action was triggered, not only the main branch. So Noetic CI for the `noetic` branch and PRs targeting it should still work.

**Comment by Cakem1x** [comment]
> > Current job runtime is 10min. I think we can improve this, e.g. via caching or by tweaking the parameters, but I suggest doing that at a later in a separate PR if the need arises.
> 
> Wow, that's long. But ok for now.

Yes, I think we should look into
* GitHub Action caching
* Building only `mesh_tools` pkgs and its source deps
  * on CLI, you could do `colcon build --packages-up-to $(colcon list --base-paths $PATH_TO_MESH_TOOLS_REPO). Maybe there is an option that does a similar thing in the ci action
* Testing only `mesh_tools` pkgs (and not its source deps)
  * e.g. via `colcon test --packages-select $(colcon list --base-paths $PATH_TO_MESH_TOOLS_REPO).  Maybe there is an option that does a similar thing in the ci action

**Pull request merged**